src/zenml/orchestrators/base_orchestrator.py

```diff
@@ -505,10 +505,12 @@ def _configure_node_context(
             runtime_config_copy, sort_keys=True, default=pydantic_encoder
         )
 
+        docker_config_json = pipeline.docker_configuration.json(sort_keys=True)
```

Comment:
Just out of curiosity, I see two different paradigms here. While the `runtime_configuration` is given to this function by the orchestrator, the `docker_configuration` is treated as a class variable of the pipeline. Is there a specific reason why it is like this?

Reply:
Mhm correct, I think this is mainly due to the `docker_configuration` being specified on the `@pipeline` decorator right now, while the runtime config is passed to `.run(...)`. However I think the entire configurations are quite messy at the moment, hopefully the design document we're creating this week will bring some more clarity in how we handle these things in the future :)

Reply:
Looking forward to it :)